crates/oxc_linter/src/rules/no_empty.rs

```diff
@@ -0,0 +1,110 @@
+use oxc_ast::{AstKind, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-empty): Disallow empty block statements")]
+#[diagnostic(severity(warning), help("Add comment inside empty {0} statement"))]
+struct NoEmptyDiagnostic(&'static str, #[label("Empty {0} statement")] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoEmpty {
+    allow_empty_catch: bool,
+}
+
+const RULE_NAME: &str = "no-empty";
+
+impl Rule for NoEmpty {
+    const NAME: &'static str = RULE_NAME;
+
+    fn from_json(value: serde_json::Value) -> Self {
+        let obj = value.get(0);
+        Self {
+            allow_empty_catch: obj
+                .and_then(|v| v.get("allowEmptyCatch"))
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or_default(),
+        }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.get().kind() {
+            AstKind::BlockStatement(block) if block.body.is_empty() => {
+                if ctx.semantic().trivias().has_comments_between(block.span) {
+                    return;
+                }
+                ctx.diagnostic(NoEmptyDiagnostic("block", block.span));
+            }
+            // The visitor does not visit the `BlockStatement` inside the `CatchClause`.
+            // See `Visit::visit_catch_clause`.
+            AstKind::CatchClause(catch_clause)
+                if !self.allow_empty_catch && catch_clause.body.body.is_empty() =>
+            {
+                if ctx.semantic().trivias().has_comments_between(catch_clause.body.span) {
```

Comment:
Do I understand correctly that if the `body` of a `CatchClause` has comments, and you'd check the `body` to find comments, you won't find them as AST Nodes?

Reply:
Yeah the AST slightly different, and the visitor is not visiting the block body.